### DISCUSSION TITLE
Power order for Relays and PWM (and Display power)

### DISCUSSION THREAD

**Discussion by pkkrusty** [arendst/Tasmota#21339] [Q&A] [answered]
I've run into this a few times, and wanted to know if anyone has seen this in documentation.

I have a ESP8266 with two PWM channels defined, PWM1 is a light and PWM2 is a fan. Works fine.  Then I decide to add an IR LED to send commands to a nearby device, so I define a dummy relay (Relay/3) in GPIO assignments.

Great, now my GUI shows PWM1 (Light) and PWM2 (Fan) and Relay 3 (Dummy).  My rules do an IRSEND on power3#state changes. 

But when I issue the command `Power3 1` the fan turns on. When I click PWM2 button, the light turns on.

Turns out the PWM gets pushed to the bottom and whatever relays take their place at the front of the line.  So the dummy relay is actually the first button on the GUI, and Light is number 2, Fan is number 3. 

I had the same issue when I added a relay to a setup with a seven segment display.  The power button assigned to the display gets pushed to the bottom, after any relays, which can screw up ordering/ webbutton defines, Alexa control...

I'd like to get this documented somewhere obvious so people can find an answer when they get confused by the inconsistency like I did. 

Thoughts?

**Comment by barbudor** [comment, marked as answer]
You should not add Relay_3 but Relay_1. Always start the index of each component at 1

Relays are enumerated before PWM so
Relay_1 => Power 1
PWM_1 => Power 2
PWM_2 => Power 3

**Reply by pkkrusty** [reply]
Yeah, defining Relay 1 or Relay 3 made no difference, it assigned it the first button and Power1. So makes sense to start a 1 for each component.

Do you know if Display Power is enumerated before or after PWM?

**Reply by sfromis** [reply]
Displays are always last, AFAIK. And you cannot influence the ordering between types.
